src/ripple/app/paths/PathFindTrustLine.h

```diff
@@ -42,19 +42,18 @@ namespace ripple {
     tens of millions of instances of this class. When modifying this class think
     carefully about the memory implications.
 */
-class PathFindTrustLine final : public CountedObject<PathFindTrustLine>
+class TrustLineBase
 {
-public:
-    PathFindTrustLine() = delete;
-
-    static std::optional<PathFindTrustLine>
-    makeItem(AccountID const& accountID, std::shared_ptr<SLE const> const& sle);
-
-    // Must be public, for make_shared
-    PathFindTrustLine(
+protected:
+    // This class should not be instantiated directly. Use one of the derived
+    // classes.
```

Comment:
This comment can be more strongly enforced by making the special members protected:
```c++
    ~TrustLineBase() = default;
    TrustLineBase(TrustLineBase const&) = default;
    TrustLineBase& operator=(TrustLineBase const&) = delete;
    TrustLineBase(TrustLineBase &&) = default;
```
Plus this keeps clients from slicing to a `TrustLineBase&`.  I omitted the deleted move assignment operator as that seemed redundant to me.  But add it in if you think it reads better.  And I added the deleted copy assignment operator (redundantly) otherwise the reader has to inspect the data members to discover that it is deleted.

Reply:
Thanks! Fixed in `bde04717c4 [fold] Additional protection agains instantiating TrustLineBase`